RubberduckTests/Symbols/DeclarationExtensionTests.cs

```diff
@@ -1,145 +1,655 @@
+//End Property
+
+//Public Property Let TestInterface_Foo(rhs As Variant)
+//End Property
+//";
+//            var vbe = new MockVbeBuilder()
+//                .ProjectBuilder("UnderTest", ProjectProtection.Unprotected)
+//                .AddComponent("TestInterface", ComponentType.ClassModule, intrface, new Selection(1, 1))
+//                .AddComponent("TestImplementation", ComponentType.ClassModule, implementation, new Selection(1, 1))
+//                .AddProjectToVbeBuilder()
+//                .Build();
+
+//            var parser = MockParser.Create(vbe.Object);
+//            parser.Parse(new CancellationTokenSource());
+
+//            var declarations = parser.State.DeclarationFinder.AllDeclarations.ToList();
+//            var member = declarations.Single(decl => decl.IdentifierName.Equals("Foo") && decl.DeclarationType == DeclarationType.Variable);
+//            var implementing = declarations.Single(decl => decl.IdentifierName.Equals("TestInterface_Foo") && decl.DeclarationType == DeclarationType.PropertyLet);
+
+//            Assert.IsFalse(implementing.ImplementsInterfaceMember(member));
+//        }
+
+//        [Test]
+//        [Category("Resolver")]
+//        public void ImplementsInterfaceMemberVariantMatchesLetAndSet()
+//        {
+//            var intrface =
+//                @"Option Explicit
+
+//Public Foo As Variant";
+
+//            var implementation =
+//                @"Option Explicit
+
+//Implements TestInterface
+
+//Public Property Get TestInterface_Foo() As Variant
+//End Property
+
+//Public Property Let TestInterface_Foo(rhs As Variant)
+//End Property
+
+//Public Property Set TestInterface_Foo(rhs As Variant)
+//End Property
+//";
+//            var vbe = new MockVbeBuilder()
+//                .ProjectBuilder("UnderTest", ProjectProtection.Unprotected)
+//                .AddComponent("TestInterface", ComponentType.ClassModule, intrface, new Selection(1, 1))
+//                .AddComponent("TestImplementation", ComponentType.ClassModule, implementation, new Selection(1, 1))
+//                .AddProjectToVbeBuilder()
+//                .Build();
+
+//            var parser = MockParser.Create(vbe.Object);
+//            parser.Parse(new CancellationTokenSource());
+
+//            var declarations = parser.State.DeclarationFinder.AllDeclarations.ToList();
+//            var member = declarations.Single(decl => decl.IdentifierName.Equals("Foo") && decl.DeclarationType == DeclarationType.Variable);
+//            var setter = declarations.Single(decl => decl.IdentifierName.Equals("TestInterface_Foo") && decl.DeclarationType == DeclarationType.PropertySet);
+//            var letter = declarations.Single(decl => decl.IdentifierName.Equals("TestInterface_Foo") && decl.DeclarationType == DeclarationType.PropertyLet);
+
+//            Assert.IsTrue(setter.ImplementsInterfaceMember(member));
+//            Assert.IsTrue(letter.ImplementsInterfaceMember(member));
+//        }
+//    }
+//}
```

Comment:
you are fixing those, right? :smile:

Reply:
Yep - I'm refactoring them to use the `DeclarationFinder` instead. That's coming later today.